### PR TITLE
feat: allow passing `DBCreateOptions` to `IdbStorage` constructor

### PR DESCRIPTION
**Pull request by hpeebles**
# Description

Each `IdbStorage` instance wraps an `IdbKeyVal` instance which itself can be configured via the `DBCreateOptions`.
Prior to this PR there are no configuration options when creating `IdbStorage` instances.
This PR makes it such that `IdbStorage` instances can optionally be configured by passing `DBCreateOptions` in the constructor which will feed through to the inner `IdbKeyVal`.

# How Has This Been Tested?

It has only been tested via the automated tests.

# Checklist:

- [x] My changes follow the guidelines in [CONTRIBUTING.md](https://github.com/dfinity/agent-js/blob/main/CONTRIBUTING.md).
- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I have edited the CHANGELOG accordingly.
- [ ] I have made corresponding changes to the documentation.

### CODE REVIEW

File: packages/auth-client/src/storage.ts [hunk 1]
```diff
@@ -70,6 +70,9 @@ export class LocalStorage implements AuthClientStorage {
  * @see implements {@link AuthClientStorage}
  */
 export class IdbStorage implements AuthClientStorage {
+  constructor(private options?: DBCreateOptions) {
```
Comment (krpeacock) [inline packages/auth-client/src/storage.ts:73]:
```suggestion
  #options: DBCreateOptions;

  /**
   * @param options - DBCreateOptions
   * @param options.dbName - name for the indexeddb database
   * @param options.storeName - name for the indexeddb Data Store
   * @param options.version - version of the database. Increment to safely upgrade
   * @constructs an {@link IdbStorage}
   * @example
   * ```typescript
   * const storage = new IdbStorage({ dbName: 'my-db', storeName: 'my-store', version: 2 });
   * ```
   */
  constructor(options?: DBCreateOptions) {
    this.#options = options ?? {};
```

File: packages/auth-client/src/storage.ts [hunk 2]
```diff
@@ -78,7 +81,7 @@ export class IdbStorage implements AuthClientStorage {
         resolve(this.initializedDb);
         return;
       }
-      IdbKeyVal.create({ version: DB_VERSION }).then(db => {
+      IdbKeyVal.create(this.options).then(db => {
```
Comment (krpeacock) [inline packages/auth-client/src/storage.ts:84]:
```suggestion
      IdbKeyVal.create(this.#options).then(db => {
```

**Review by krpeacock — CHANGES_REQUESTED** [review]
Suggestions:
- prefer [private properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties) over typescript `private` keyword
- JSDocs for constructor

**Comment by hpeebles** [comment]
> Suggestions:
> 
> * prefer [private properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_properties) over typescript `private` keyword
> * JSDocs for constructor

Nice! I'd never come across these before. I've merged in your suggestions.

**Review by krpeacock — APPROVED** [review]

**Pull request merged**